### ISSUE TITLE
compiled application fails with "This application failed to start because it could not find or load the Qt platform plugin "xcb"."

### ISSUE THREAD

**Issue by matkoniecz** [OpenOrienteering/mapper#644]
I used `sudo make install` as recommended method failed (#643)

Now application crashes instead of starting with:

> This application failed to start because it could not find or load the Qt platform plugin "xcb".
> 
> Available platform plugins are: xcb.
> 
> Reinstalling the application may fix this problem.
> Aborted (core dumped)

Installing xcb package changed nothing.

Compiled on 32 bit Ubuntu 14.04.4 LTS.

**Comment by matkoniecz** [comment]
And it also breaks Mapper installed from openorienteering-mapper package...

**Comment by dg0yt** [comment]
If you ```sudo make install```, it will overwrite existing files. You will need to reinstall the openorienteering-mapper package to repair this.

By default, we build a more recent Qt (which will take a while) than what your distribution provides. In the build tree, the Mapper executable will probably start without errror. But ```make install``` changes the library paths in the executable, so the installed file will not (without some help) 
find the Qt libraries and plugins in the build tree.

To use the distribution's Qt, install the neccessary development packages, and pass ```-DMapper_BUILD_QT=0``` (and maybe some more) to cmake.

Maybe you want to study debian.rules and debian.control on https://build.opensuse.org/package/show/home:dg0yt/openorienteering-mapper-unstable to understand the current build rules and dependencies for DEB systems.

**Comment by yevhenmazur** [comment]
I've faced with the same situation on my Gentoo setup. As a workaround I had to create couple of symlinks, like these:
```
/usr/lib/libQt5XcbQpa.so.5 -> /home/yudjin/git/mapper/build/3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/lib/libQt5XcbQpa.so.5
/usr/lib/libQt5XcbQpa.so.5.5 -> /home/yudjin/git/mapper/build/3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/lib/libQt5XcbQpa.so.5.5
/usr/lib/libQt5XcbQpa.so.5.5.1 -> /home/yudjin/git/mapper/build/3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/lib/libQt5XcbQpa.so.5.5.1
```
I know, it is not true way, but it is works.

**Comment by dg0yt** [comment]
The typical solution is not adding symlinks in system directories, but adjusting LD_LIBRARY_PATH and maybe more environment variables in a wrapper script. (And that script could call the executable in the build directory, if it is just for your personal use.)

**Comment by matkoniecz** [comment]
> You will need to reinstall the openorienteering-mapper package to repair this.

Well, that failed to fix anything (it was the first thing that I tried).

**Comment by dg0yt** [comment]
Maybe you should first report what you already tried and how it failed, instead of letting me guess piece by piece, in case you need help to repair the installed mapper package.

**Comment by matkoniecz** [comment]
> Maybe you should first report what you already tried and how it failed, instead of letting me guess piece by piece, in case you need help to repair the installed mapper package.

```
sudo apt-get purge openorienteering-mapper 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libpolyclipping16 libqt5clucene5 libqt5designer5 libqt5designercomponents5
  libqt5help5 libqt5printsupport5 libqt5sql5 libqt5sql5-sqlite libqt5xml5
  qt5-image-formats-plugins qttools5-dev-tools qttranslations5-l10n
Use 'apt-get autoremove' to remove them.
The following packages will be REMOVED:
  openorienteering-mapper*
0 upgraded, 0 newly installed, 1 to remove and 1 not upgraded.
After this operation, 12,8 MB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 1115747 files and directories currently installed.)
Removing openorienteering-mapper (0.6.1-4.1) ...
dpkg: warning: while removing openorienteering-mapper, directory '/usr/share/openorienteering-mapper/symbol sets/15000' not empty so not removed
dpkg: warning: while removing openorienteering-mapper, directory '/usr/share/openorienteering-mapper/symbol sets/10000' not empty so not removed
dpkg: warning: while removing openorienteering-mapper, directory '/usr/share/openorienteering-mapper/symbol sets/5000' not empty so not removed
dpkg: warning: while removing openorienteering-mapper, directory '/usr/share/doc/openorienteering-mapper' not empty so not removed
Processing triggers for hicolor-icon-theme (0.13-1) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Processing triggers for gnome-menus (3.10.1-0ubuntu2) ...
Processing triggers for mime-support (3.54ubuntu1.1) ...
Processing triggers for desktop-file-utils (0.22-1ubuntu1) ...
```

and

```
sudo apt-get autoremove 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages will be REMOVED:
  libpolyclipping16 libqt5clucene5 libqt5designer5 libqt5designercomponents5
  libqt5help5 libqt5printsupport5 libqt5sql5 libqt5sql5-sqlite libqt5xml5
  qt5-image-formats-plugins qttools5-dev-tools qttranslations5-l10n
0 upgraded, 0 newly installed, 12 to remove and 1 not upgraded.
After this operation, 25,3 MB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 1115688 files and directories currently installed.)
Removing libpolyclipping16 (6.1.3a-2) ...
Removing qttools5-dev-tools (5.2.1-8build1) ...
Removing libqt5help5:i386 (5.2.1-8build1) ...
Removing libqt5clucene5:i386 (5.2.1-8build1) ...
Removing libqt5designercomponents5:i386 (5.2.1-8build1) ...
Removing libqt5designer5:i386 (5.2.1-8build1) ...
Removing libqt5printsupport5:i386 (5.2.1+dfsg-1ubuntu14.3) ...
Removing libqt5sql5-sqlite:i386 (5.2.1+dfsg-1ubuntu14.3) ...
Removing libqt5sql5:i386 (5.2.1+dfsg-1ubuntu14.3) ...
Removing libqt5xml5:i386 (5.2.1+dfsg-1ubuntu14.3) ...
Removing qt5-image-formats-plugins:i386 (5.2.1-1) ...
Removing qttranslations5-l10n (5.2.1-1) ...
Processing triggers for libc-bin (2.19-0ubuntu6.7) ...
```

what removed Mapper program from system

```
Mapper
No command 'Mapper' found, did you mean:
 Command 'mapper' from package 'dnssec-tools' (universe)
 Command 'rapper' from package 'raptor2-utils' (main)
 Command 'rapper' from package 'raptor-utils' (main)
 Command 'apper' from package 'apper' (universe)
Mapper: command not found
```

followed by reinstallation

```
sudo apt-get install openorienteering-mapper 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following extra packages will be installed:
  libpolyclipping16 libqt5clucene5 libqt5designer5 libqt5designercomponents5
  libqt5help5 libqt5printsupport5 libqt5sql5 libqt5sql5-sqlite libqt5xml5
  qt5-image-formats-plugins qttools5-dev-tools qttranslations5-l10n
The following NEW packages will be installed:
  libpolyclipping16 libqt5clucene5 libqt5designer5 libqt5designercomponents5
  libqt5help5 libqt5printsupport5 libqt5sql5 libqt5sql5-sqlite libqt5xml5
  openorienteering-mapper qt5-image-formats-plugins qttools5-dev-tools
  qttranslations5-l10n
0 upgraded, 13 newly installed, 0 to remove and 1 not upgraded.
Need to get 0 B/10,4 MB of archives.
After this operation, 38,1 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
WARNING: The following packages cannot be authenticated!
  openorienteering-mapper
Install these packages without verification? [y/N] y
Selecting previously unselected package libqt5clucene5:i386.
(Reading database ... 1115446 files and directories currently installed.)
Preparing to unpack .../libqt5clucene5_5.2.1-8build1_i386.deb ...
Unpacking libqt5clucene5:i386 (5.2.1-8build1) ...
Selecting previously unselected package libqt5xml5:i386.
Preparing to unpack .../libqt5xml5_5.2.1+dfsg-1ubuntu14.3_i386.deb ...
Unpacking libqt5xml5:i386 (5.2.1+dfsg-1ubuntu14.3) ...
Selecting previously unselected package libqt5designer5:i386.
Preparing to unpack .../libqt5designer5_5.2.1-8build1_i386.deb ...
Unpacking libqt5designer5:i386 (5.2.1-8build1) ...
Selecting previously unselected package libqt5designercomponents5:i386.
Preparing to unpack .../libqt5designercomponents5_5.2.1-8build1_i386.deb ...
Unpacking libqt5designercomponents5:i386 (5.2.1-8build1) ...
Selecting previously unselected package libqt5sql5:i386.
Preparing to unpack .../libqt5sql5_5.2.1+dfsg-1ubuntu14.3_i386.deb ...
Unpacking libqt5sql5:i386 (5.2.1+dfsg-1ubuntu14.3) ...
Selecting previously unselected package libqt5help5:i386.
Preparing to unpack .../libqt5help5_5.2.1-8build1_i386.deb ...
Unpacking libqt5help5:i386 (5.2.1-8build1) ...
Selecting previously unselected package libqt5printsupport5:i386.
Preparing to unpack .../libqt5printsupport5_5.2.1+dfsg-1ubuntu14.3_i386.deb ...
Unpacking libqt5printsupport5:i386 (5.2.1+dfsg-1ubuntu14.3) ...
Selecting previously unselected package libqt5sql5-sqlite:i386.
Preparing to unpack .../libqt5sql5-sqlite_5.2.1+dfsg-1ubuntu14.3_i386.deb ...
Unpacking libqt5sql5-sqlite:i386 (5.2.1+dfsg-1ubuntu14.3) ...
Selecting previously unselected package qt5-image-formats-plugins:i386.
Preparing to unpack .../qt5-image-formats-plugins_5.2.1-1_i386.deb ...
Unpacking qt5-image-formats-plugins:i386 (5.2.1-1) ...
Selecting previously unselected package libpolyclipping16.
Preparing to unpack .../libpolyclipping16_6.1.3a-2_i386.deb ...
Unpacking libpolyclipping16 (6.1.3a-2) ...
Selecting previously unselected package openorienteering-mapper.
Preparing to unpack .../openorienteering-mapper_0.6.1-4.1_i386.deb ...
Unpacking openorienteering-mapper (0.6.1-4.1) ...
Selecting previously unselected package qttools5-dev-tools.
Preparing to unpack .../qttools5-dev-tools_5.2.1-8build1_i386.deb ...
Unpacking qttools5-dev-tools (5.2.1-8build1) ...
Selecting previously unselected package qttranslations5-l10n.
Preparing to unpack .../qttranslations5-l10n_5.2.1-1_all.deb ...
Unpacking qttranslations5-l10n (5.2.1-1) ...
Processing triggers for gnome-menus (3.10.1-0ubuntu2) ...
Processing triggers for mime-support (3.54ubuntu1.1) ...
Processing triggers for desktop-file-utils (0.22-1ubuntu1) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Processing triggers for hicolor-icon-theme (0.13-1) ...
Setting up libqt5clucene5:i386 (5.2.1-8build1) ...
Setting up libqt5xml5:i386 (5.2.1+dfsg-1ubuntu14.3) ...
Setting up libqt5designer5:i386 (5.2.1-8build1) ...
Setting up libqt5designercomponents5:i386 (5.2.1-8build1) ...
Setting up libqt5sql5:i386 (5.2.1+dfsg-1ubuntu14.3) ...
Setting up libqt5help5:i386 (5.2.1-8build1) ...
Setting up libqt5printsupport5:i386 (5.2.1+dfsg-1ubuntu14.3) ...
Setting up libqt5sql5-sqlite:i386 (5.2.1+dfsg-1ubuntu14.3) ...
Setting up qt5-image-formats-plugins:i386 (5.2.1-1) ...
Setting up libpolyclipping16 (6.1.3a-2) ...
Setting up openorienteering-mapper (0.6.1-4.1) ...
Setting up qttools5-dev-tools (5.2.1-8build1) ...
Setting up qttranslations5-l10n (5.2.1-1) ...
Processing triggers for libc-bin (2.19-0ubuntu6.7) ...
```

what is not changing problem

```
Mapper
This application failed to start because it could not find or load the Qt platform plugin "xcb".

Available platform plugins are: xcb.

Reinstalling the application may fix this problem.
Aborted (core dumped)
```

**Comment by dg0yt** [comment]
```make install``` may have created a ```/usr/bin/qt.conf``` which changes the plugin path. Remove this file if it exists.

If this doesn't help: What is the output of ```dpkg-query -l libqt5*5```? You may have overwritten other parts of Qt.

**Comment by matkoniecz** [comment]
/usr/bin/qt.conf is not existing, removing /usr/bin/qtconfig resulted in no noticeable changes

`dpkg-query -l libqt5*5` gives

```Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                                  Version                         Architecture                    Description
+++-=====================================================-===============================-===============================-===============================================================================================================
ii  libqt5clucene5:i386                                   5.2.1-8build1                   i386                            Qt 5 CLucene module
un  libqt5core5                                           <none>                          <none>                          (no description available)
ii  libqt5dbus5:i386                                      5.2.1+dfsg-1ubuntu14.3          i386                            Qt 5 D-Bus module
ii  libqt5designer5:i386                                  5.2.1-8build1                   i386                            Qt 5 designer module
ii  libqt5designercomponents5:i386                        5.2.1-8build1                   i386                            Qt 5 Designer components module
ii  libqt5gui5:i386                                       5.2.1+dfsg-1ubuntu14.3          i386                            Qt 5 GUI module
ii  libqt5help5:i386                                      5.2.1-8build1                   i386                            Qt 5 help module
ii  libqt5multimedia5:i386                                5.2.1-0ubuntu5                  i386                            Qt 5 Multimedia module
ii  libqt5network5:i386                                   5.2.1+dfsg-1ubuntu14.3          i386                            Qt 5 network module
ii  libqt5opengl5:i386                                    5.2.1+dfsg-1ubuntu14.3          i386                            Qt 5 OpenGL module
ii  libqt5printsupport5:i386                              5.2.1+dfsg-1ubuntu14.3          i386                            Qt 5 print support module
ii  libqt5sql5:i386                                       5.2.1+dfsg-1ubuntu14.3          i386                            Qt 5 SQL module
ii  libqt5widgets5:i386                                   5.2.1+dfsg-1ubuntu14.3          i386                            Qt 5 widgets module
ii  libqt5xml5:i386                                       5.2.1+dfsg-1ubuntu14.3          i386                            Qt 5 XML module
```

**Comment by dg0yt** [comment]
I didn't ask to remove qtconfig... 

```sudo debsums -c libqt5core5a libqt5widgets5 libqt5gui5 libqt5clucene5 libqt5help5 libqt5network5 libqt5opengl5 libqt5printsupport5 libqt5sql5``` should output installed files which differ from the given packages. 

```sudo apt-get install --reinstall PACKAGE_NAME``` should reinstall a broken package identified in the previous step.

(from http://askubuntu.com/questions/57682/find-and-reinstall-packages-with-corrupted-files-without-breaking-anything)

**Comment by matkoniecz** [comment]
`sudo debsums -c libqt5core5a libqt5widgets5 libqt5gui5 libqt5clucene5 libqt5help5 libqt5network5 libqt5opengl5 libqt5printsupport5 libqt5sql5` failed to identify any broken packages.
I tried 
`sudo debsums -ca libqt5core5a libqt5widgets5 libqt5gui5 libqt5clucene5 libqt5help5 libqt5network5 libqt5opengl5 libqt5printsupport5 libqt5sql5` but it also failed to find anything.

**Comment by dg0yt** [comment]
Is it really ```/usr/bin/Mapper``` which is failing? If yes, ```strace -e trace=file /usr/bin/Mapper 2>&1 | grep qt``` will output which files named like qt are accessed when the program starts.

**Comment by matkoniecz** [comment]
```
trace -e trace=file /usr/bin/Mapper 2>&1 | grep qt
stat64("/home/mateusz/.config/QtProject/qtlogging.ini", 0xbf860b20) = -1 ENOENT (No such file or directory)
stat64("/etc/xdg/lubuntu/QtProject/qtlogging.ini", 0xbf860b20) = -1 ENOENT (No such file or directory)
stat64("/etc/xdg/xdg-Lubuntu/QtProject/qtlogging.ini", 0xbf860b20) = -1 ENOENT (No such file or directory)
stat64("/usr/share/upstart/xdg/QtProject/qtlogging.ini", 0xbf860b20) = -1 ENOENT (No such file or directory)
stat64("/etc/xdg/QtProject/qtlogging.ini", 0xbf860b20) = -1 ENOENT (No such file or directory)
stat64("/usr/bin/qt.conf", 0xbf8609f0)  = -1 ENOENT (No such file or directory)
stat64("/usr/bin/qt.conf", 0xbf8604d0)  = -1 ENOENT (No such file or directory)
```

**Comment by matkoniecz** [comment]
Running `/usr/bin/Mapper` directly also gives

```
This application failed to start because it could not find or load the Qt platform plugin "xcb".

Available platform plugins are: xcb.

Reinstalling the application may fix this problem.
Aborted (core dumped)
```

**Comment by dg0yt** [comment]
That's still confusing:
- There is a xcb plugin but doesn't get loaded. This indicates a version mismatch. 
- Mapper doesn't touch your system's qt5/plugins folder (where the xcb plugin resides) but it does so on my computer.
- Mapper tries to read a qtlogging.ini, but it doesn't do so on my PC. qtlogging.ini was introduced in Qt 5.3, according to http://blog.qt.io/blog/2014/03/11/qt-weekly-1-categorized-logging/.

This indicates that /usr/bin/Mapper is linked with (and loading) Qt 5.3 libraries. But the installed version is 5.2.1 in Ubuntu 14.04. Maybe you failed to properly reinstall the openorienteering-mapper package?

```dpkg-query -l openorienteering-mapper```
```sudo debsums -c openorienteering-mapper```

(And of course you can get much more output from strace by dropping the pipe to ```grep qt```.)

**Comment by matkoniecz** [comment]
` sudo debsums -c openorienteering-mapper` gives empty output and `dpkg-query -l openorienteering-mapper` gives:

```
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                       Version                    Architecture               Description
+++-==========================================-==========================-==========================-==========================================================================================
ii  openorienteering-mapper                    0.6.1-4.1                  i386                       Orienteering map drawing software
```

full strace output is at https://gist.github.com/matkoniecz/e84f9d9a18ef84c60ba65c79fb09d2bc

I already tried `sudo apt-get install --reinstall openorienteering-mapper` and I did it again, without noticeable changes.

**Comment by dg0yt** [comment]
Mapper 0.6.1-4.1 is the current stable build. But strace shows that Qt is loaded from /usr/lib/openorienteering-mapper/lib. This location was probably created during ```sudo make install```. Please remove it.

**Comment by matkoniecz** [comment]
Now it works, big thanks for help!

I am unsure whatever it should be closed - `sudo make install ` is standard installation method, but not mentioned as recommended in documentation...

**Comment by dg0yt** [comment]
Thank you for the feedback. From my development environment it is very hard to cover all issues which can appear in different configurations.

I will probably change the defaults for Linux to not build anything but Mapper. That's what is used for packaging, and only developers should enable different options. Documentation needs to be updated accordingly.

Apart from this, the issue indicates that there is a qt.conf embedded in the Mapper executable, pointing to /usr/lib/openorienteering-mapper/lib. This needs to be removed from our release packages.

**Comment by dg0yt** [comment]
According to the build logs, there is no embedded qt.conf in the package from OBS. I don't know why Mapper picked up the non-standard libraries.

**Comment by dg0yt** [comment]
In the meantime I run into this issue, too. For some reason, Mapper kept embedding a qt.conf even after a configuration change where it shouldn't. However, a clean build from scratch should not have be affected.

**Comment by bagage** [comment]
I'm running into this issue as well, but the error is due to `Mapper` linking to the wrong xcb library. 

I am trying to build it with `cmake ..  -DMapper_BUILD_QT=ON`.

Qt5 is *not* installed, see `dpkg -l | grep qt.*5` output:

```
ii  libqt5core5a:amd64                    5.7.1+dfsg-1                      amd64        Qt 5 core module
ii  libqt5dbus5:amd64                     5.7.1+dfsg-1                      amd64        Qt 5 D-Bus module
ii  libqt5gui5:amd64                      5.7.1+dfsg-1                      amd64        Qt 5 GUI module
ii  libqt5network5:amd64                  5.7.1+dfsg-1                      amd64        Qt 5 network module
ii  libqt5opengl5:amd64                   5.7.1+dfsg-1                      amd64        Qt 5 OpenGL module
ii  libqt5printsupport5:amd64             5.7.1+dfsg-1                      amd64        Qt 5 print support module
ii  libqt5qml5:amd64                      5.7.1-1                           amd64        Qt 5 QML module
ii  libqt5quick5:amd64                    5.7.1-1                           amd64        Qt 5 Quick library
ii  libqt5scintilla2-12v5                 2.9.3+dfsg-4                      amd64        Qt5 port of the Scintilla source code editing widget
ii  libqt5scintilla2-l10n                 2.9.3+dfsg-4                      all          Scintilla source code editing widget for Qt5, translation files
ii  libqt5sql5:amd64                      5.7.1+dfsg-1                      amd64        Qt 5 SQL module
ii  libqt5sql5-sqlite:amd64               5.7.1+dfsg-1                      amd64        Qt 5 SQLite 3 database driver
ii  libqt5svg5:amd64                      5.7.1~20161021-2                  amd64        Qt 5 SVG module
ii  libqt5test5:amd64                     5.7.1+dfsg-1                      amd64        Qt 5 test module
ii  libqt5webkit5:amd64                   5.7.1+dfsg-1                      amd64        Web content engine library for Qt
ii  libqt5widgets5:amd64                  5.7.1+dfsg-1                      amd64        Qt 5 widgets module
ii  libqt5x11extras5:amd64                5.7.1~20161021-2                  amd64        Qt 5 X11 extras
ii  qt5-doc                               5.7.1-1                           all          Qt 5 API Documentation
ii  qt5-gtk-platformtheme:amd64           5.7.1+dfsg-1                      amd64        Qt 5 GTK+ 3 platform theme
ii  qt5-qmake:amd64                       5.7.1+dfsg-1                      amd64        Qt 5 qmake Makefile generator tool
ii  qtbase5-doc                           5.7.1+dfsg-1                      all          Qt 5 base documentation
ii  qtconnectivity5-doc                   5.7.1~20161021-2                  all          Qt 5 Sensors documentation
ii  qtdeclarative5-doc                    5.7.1-1                           all          Qt 5 declarative documentation
ii  qtgraphicaleffects5-doc               5.7.1~20161021-3                  all          Qt 5 graphical effects documentation
ii  qtlocation5-doc                       5.7.1-1                           all          Qt 5 Positioning documentation
ii  qtmultimedia5-doc                     5.7.1~20161021-2                  all          Qt 5 multimedia documentation
ii  qtquickcontrols5-doc                  5.7.1~20161021-2                  all          Qt 5 Quick Controls documentation
ii  qtscript5-doc                         5.7.1~20161021+dfsg-2             all          Qt 5 script documentation
ii  qtsensors5-doc                        5.7.1~20161021-2                  all          Qt 5 Sensors documentation
ii  qtserialport5-doc                     5.7.1~20161021-2                  all          Qt 5 serial port documentation
ii  qtsvg5-doc                            5.7.1~20161021-2                  all          Qt 5 SVG documentation
ii  qttools5-doc                          5.7.1-1                           all          Qt 5 tools documentation
ii  qttranslations5-l10n                  5.7.1~20161021-1                  all          translations for Qt 5
ii  qtwebkit5-doc                         5.7.1+dfsg-1                      all          Qt 5 webkit documentation
ii  qtwebkit5-examples-doc                5.7.1~20161021-2                  all          Qt 5 webkit examples documentation
ii  qtwebsockets5-doc                     5.7.1~20161021-4                  all          Qt 5 Web Sockets documentation
ii  qtx11extras5-doc                      5.7.1~20161021-2                  all          Qt 5 X11 extras documentation
ii  qtxmlpatterns5-doc                    5.7.1~20161021-3                  all          Qt 5 XML patterns documentation
```

But when I run `ldd build/src/Mapper` I get:

```
	linux-vdso.so.1 (0x00007ffc007c5000)
	libQt5Network.so.5 => /home/gautier/code/openorienteeringmapper/build/3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/lib/libQt5Network.so.5 (0x00007f8d930aa000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f8d92e71000)
	libproj.so.12 => /usr/lib/x86_64-linux-gnu/libproj.so.12 (0x00007f8d92c07000)
	libQt5Xml.so.5 => /home/gautier/code/openorienteeringmapper/build/3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/lib/libQt5Xml.so.5 (0x00007f8d92bcc000)
	libQt5PrintSupport.so.5 => /home/gautier/code/openorienteeringmapper/build/3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/lib/libQt5PrintSupport.so.5 (0x00007f8d92b59000)
	libQt5Widgets.so.5 => /home/gautier/code/openorienteeringmapper/build/3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/lib/libQt5Widgets.so.5 (0x00007f8d92516000)
	libQt5Gui.so.5 => /home/gautier/code/openorienteeringmapper/build/3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/lib/libQt5Gui.so.5 (0x00007f8d91fa1000)
	libQt5Core.so.5 => /home/gautier/code/openorienteeringmapper/build/3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/lib/libQt5Core.so.5 (0x00007f8d91ae2000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f8d91760000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f8d9145c000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f8d91243000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f8d90ea5000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f8d90c88000)
	libGL.so.1 => /usr/lib/x86_64-linux-gnu/libGL.so.1 (0x00007f8d90a16000)
	libpng16.so.16 => /usr/lib/x86_64-linux-gnu/libpng16.so.16 (0x00007f8d907e3000)
	libicui18n.so.57 => /usr/lib/x86_64-linux-gnu/libicui18n.so.57 (0x00007f8d90367000)
	libicuuc.so.57 => /usr/lib/x86_64-linux-gnu/libicuuc.so.57 (0x00007f8d8ffbf000)
	libicudata.so.57 => /usr/lib/x86_64-linux-gnu/libicudata.so.57 (0x00007f8d8e542000)
	libpcre16.so.3 => /usr/lib/x86_64-linux-gnu/libpcre16.so.3 (0x00007f8d8e2d9000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f8d8e0d5000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f8d8decd000)
	/lib64/ld-linux-x86-64.so.2 (0x000055ecd9cca000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007f8d8dca1000)
	libxcb-dri3.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri3.so.0 (0x00007f8d8da9e000)
	libxcb-present.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-present.so.0 (0x00007f8d8d89b000)
	libxcb-sync.so.1 => /usr/lib/x86_64-linux-gnu/libxcb-sync.so.1 (0x00007f8d8d694000)
	libxshmfence.so.1 => /usr/lib/x86_64-linux-gnu/libxshmfence.so.1 (0x00007f8d8d491000)
	libglapi.so.0 => /usr/lib/x86_64-linux-gnu/libglapi.so.0 (0x00007f8d8d262000)
	libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007f8d8d04e000)
	libXdamage.so.1 => /usr/lib/x86_64-linux-gnu/libXdamage.so.1 (0x00007f8d8ce4b000)
	libXfixes.so.3 => /usr/lib/x86_64-linux-gnu/libXfixes.so.3 (0x00007f8d8cc45000)
	libX11-xcb.so.1 => /usr/lib/x86_64-linux-gnu/libX11-xcb.so.1 (0x00007f8d8ca43000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007f8d8c703000)
	*libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007f8d8c4db000)*
	libxcb-glx.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-glx.so.0 (0x00007f8d8c2be000)
	libxcb-dri2.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri2.so.0 (0x00007f8d8c0b9000)
	libXxf86vm.so.1 => /usr/lib/x86_64-linux-gnu/libXxf86vm.so.1 (0x00007f8d8beb3000)
	libdrm.so.2 => /usr/lib/x86_64-linux-gnu/libdrm.so.2 (0x00007f8d8bca3000)
	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007f8d8ba9f000)
	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007f8d8b897000)
```

I guess `Mapper` should use `./build/3rd-party/qt5/Qt5-prefix/src/Qt5-build/qtbase/lib/libxcb-static.a` instead?

**Comment by dg0yt** [comment]
`./build/3rd-party/qt5/Qt5-prefix/src/Qt5-build/qtbase/lib/libxcb-static.a` would be used when linking static at build time, but `ldd` tells what is linked dynamically at runtime. 

You *do* have Qt 5.7.1 installed, but probably not the -dev packages. I wonder which platform plugin is loaded by your app. (`ldd` won't tell you.)

**Comment by bagage** [comment]
I know that `libxcb-static.a` is a static library, but that's the only one that gets builded by `mapper` so I believe this is the one mapper should use. Since I asked for Qt to be built, should not be `libxcb` built too?

Do you know any way to show platform plugin path? I tried overwriting `export QT_QPA_PLATFORM_PLUGIN_PATH=./3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/plugins/platforms ` but this lead to another error:

```
This application failed to start because it could not find or load the Qt platform plugin "xcb".

Available platform plugins are: eglfs (from ./3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/plugins/platforms), minimal (from ./3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/plugins/platforms), minimalegl (from ./3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/plugins/platforms), offscreen (from ./3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/plugins/platforms), xcb (from ./3rd-party/qt5/Qt5-prefix/usr/lib/openorienteering-mapper/plugins/platforms).

Reinstalling the application may fix this problem.
```

Did you ever build mapper on linux with `Mapper_BUILD_QT=ON` and no Qt dev libraries installed on system?

**Comment by bagage** [comment]
Ok so the issue was actually only a missing package `qtbase5-private-dev`, fixed by #821. 
Nevertheless, there are still 2 issues here:
- I think that CMakeLists.txt should turn **OFF** Mapper_BUILD_QT option by default - it's only useful for packagers, not developers.
- Even if using option Mapper_BUILD_QT, one have to install `qtbase5-private-dev`. Either Mapper_BUILT_QT should build these libraries or they should be required to be installed. The former is the preferred solution here I believe.

**Comment by dg0yt** [comment]
I do not plan to maintain the superbuilds inside the Mapper repository. There is a work-in-progress issue to modify the build system for 0.7, and there is a OpenOrienteering/superbuild repository which more or less already does what I am trying to achieve.
